startup_scripts/250_dcim_interfaces.py

```diff
@@ -0,0 +1,55 @@
+from dcim.models import Interface, Device
+from dcim.constants import IFACE_TYPE_CHOICES
+from extras.models import CustomField, CustomFieldValue
+from ruamel.yaml import YAML
+
+from pathlib import Path
+import sys
+
+file = Path('/opt/netbox/initializers/dcim_interfaces.yml')
+if not file.is_file():
+  sys.exit()
+
+with file.open('r') as stream:
+  yaml = YAML(typ='safe')
+  interfaces = yaml.load(stream)
+
+  required_assocs = {
+    'device': (Device, 'name')
+  }
+
+  if interfaces is not None:
+    for params in interfaces:
+      custom_fields = params.pop('custom_fields', None)
+
+      for assoc, details in required_assocs.items():
+        model, field = details
+        query = { field: params.pop(assoc) }
+
+        params[assoc] = model.objects.get(**query)
+
+      if 'form_factor' in params:
+        for outer_list in IFACE_TYPE_CHOICES:
+          for ffactor_choices in outer_list[1]:
+            if params['form_factor'] in ffactor_choices:
+              params['form_factor'] = ffactor_choices[0]
```

Comment:
`form_factor` does not seem to be [a field on `Interface`][iface]. I believe the field is named `type`. Maybe that should be the name in the yaml as well.

[iface]: https://github.com/netbox-community/netbox/blob/d849a49842da7157b867158bfa4cdafdf7ff9a4c/netbox/dcim/models.py#L2150-L2242

Reply:
This is a weird one. In the API it is called form_factor, and I remember a few months ago I specified it as form_factor the .yaml files and it worked. But you're right, in the model its called type, that one should be correct.